### PR TITLE
Adding a release page under "Release Information"

### PR DESCRIPTION
**Pull request by jimangel**
Fixes: https://github.com/kubernetes/website/issues/20293

> **TL;DR:** I consolidated all release information under k8s.io/releases including all general automation around content updating (based off of SIG Release's [schedule.yaml](https://raw.githubusercontent.com/kubernetes/sig-release/master/releases/schedule.yaml)).

Working with SIG Release and SIG Docs, I'm proposing a "MVP," or "Phase 1," of consolidating Kubernetes' release documentation. A wave of improvements could follow once this foundation exists.

**The ultimate goal is to raise visibility into the Kubernetes release cycle and reduce website information fragmentation.** (More history in original issue: https://github.com/kubernetes/website/issues/20293)

Given the countless ways to consume and visualize release data, this PR focuses on a few high-level goals:

- [x] Allow for SIG Docs/Release Co-ownership
- [x] Data driven (no manual maintenance)
- [x] No dual sourced files*
- [x] No broken links
- [ ] No stale release history in previous version docs (ex: https://v1-18.docs.kubernetes.io/releases)

*dual sourced as in created and maintained in two locations.

If folks agree with this approach, I recommend merging this PR, after initial review, and we can iterate on feedback after? :thumbsup: :thumbsdown: 

## General Architecture

`[k/sig-release]`--copy-->`[k/website/data]`--generate-->`[k/website/releases]`

A folder created in `content/en/releases` creates a new k8s.io/releases page.

A script `./scripts/update-release-info.sh` copies release markdown pages from SIG Release.

Data for Kubernetes versions is copied from SIG Release and stored in `data/releases/schedule.yaml`

All content copied over has a modified front matter of `type: docs` prepended in order to inherit the main "/docs" theme for consistency.

A new shortcode `{{< release-data >}}` was created to render release information from `data/releases/schedule.yaml` in `layouts/shortcodes/release-data.html`. This section could be broken into a couple `partials` or multiple shortcodes. However, I found keeping everything in a single file made it easier to connect all the parts.

`content/en/releases/_index.html` builds the entire section and pulls the latest html-ready release data using the shortcode `{{< release-data >}}`

An OWNERS file is placed in `data/releases/`, `content/en/releases/` that includes SIG Release.

Summed up:

* Copy over SIG Release markdown to k/website (`content/en/releases/`)
* Copy over SIG Release schedule data to k/website (`data/releases/schedule.yaml`)
* Generate "Release Information" under k8s.io/releases (`content/en/releases/_index.html -> layouts/shortcodes/release-data.html`)

In the future, SIG Release could move their data source to k/website and the process would be automated 100% with co-ownership between SIG Docs and SIG Release.

## Other minor changes included in this PR

- Added a link in the community page to "Releases"
- Moved /setup/releases to /releases
    - Copied file `version-skew-policy.md` (added `type: docs`)
    - `/docs/setup/version-skew-policy/     /docs/setup/release/version-skew-policy/   301` changed to `/docs/setup/version-skew-policy/     /releases/version-skew-policy/   301`
    - Created redirect `/docs/setup/release/                 /releases/   301`
    - Created redirect `/docs/setup/release/version-skew-policy/     /releases/version-skew-policy/   301`
    - Created file `notes.md` (added `type: docs`)
    - Deleted directory `/docs/setup/release/notes/`
    - Created redirect `/docs/setup/release/notes/     /releases/notes/   301`
    - Tested redirects via Netlify build
- SIG Release OWNERS updated
    - OWNERS_ALIASES updated from `git.k8s.io/release/OWNERS_ALIASES`
        - `release-engineering-approvers`
        - `release-engineering-reviewers`
    - OWNERS file added to `data/releases/`, `content/en/releases/` including the above approvers / reviewers.

## Open questions & items (non blocking)

- [ ] `versionSkew` shortcode could be improved to use `data/releases` vs. `site.Params` (and math)
- [ ] Could this fix any of the release docs process (updating the `config.toml`)?
- [ ] Any Localization adoption challenges?
- [ ] How far back should we include for history? (updating `data/releases/schedule.yaml`)
- [ ] How could this better be integrated into SIG Release? (krel / [schedule-builder](https://github.com/kubernetes/release/tree/master/cmd/schedule-builder))
- [ ] How to better consolidate the release notes vs. relnotes.k8s.io vs. ...?
- [ ] I think we could automate the release binary downloads if we could update `data/releases/schedule.yaml` with URLs (replacing what downloadkubernetes.com does)
    - [ ] Something like: https://golang.org/dl/?

## To update everything

```
chmod +x ./scripts/update-release-info.sh 
./scripts/update-release-info.sh
```

## Next steps

- [x] Add links with examples once page renders
- [x] Verify k8s.io/releases works
- [ ] ~~Does having data in /data/releases/ make the most sense? (localization?)~~
- [x] Sign off from SIG Release (:wave: @LappleApple @justaugustus @tpepper @saschagrunert)
- [x] Sign off from Docs release lead (:wave: @pi-victor)
- [ ] Sign off from majority of SIG Docs Co-Chairs & Tech leads (:wave: @irvifa @kbarnard10 @kbhawkey @onlydole @sftim)
- [ ] Update [schedule-builder](https://github.com/kubernetes/release/tree/master/cmd/schedule-builder) and other copied files with reference comments
- [ ] Add redirects to older `release-x.y` websites to current page (update release handbook too)

/kind feature
/cc @kubernetes/release-engineering @kubernetes/sig-docs-en-owners @mkorbi
/priority important-soon
/sig release
/sig docs
/language en
/hold for review

### CODE REVIEW

File: content/en/releases/patch-releases.md
Comment (mrbobbytables) [inline content/en/releases/patch-releases.md:186]: This will need to be replaced in the update-release-info script

File: content/en/releases/release.md
Comment (mrbobbytables) [inline content/en/releases/release.md:350]: This will need to be replaced in the update-release-info script

File: content/en/releases/release.md
Comment (mrbobbytables) [inline content/en/releases/release.md:357]: This will need to be replaced in the update-release-info script

**Review by mrbobbytables — COMMENTED** [review]
Quick review just highlighting some of the links that will need to be updated via the script.

File: layouts/shortcodes/release-data.html
Comment (sftim) [inline layouts/shortcodes/release-data.html:10]: We should use localized string lookups here.

File: layouts/shortcodes/release-data.html
Comment (sftim) [inline layouts/shortcodes/release-data.html:18]: We should use localized string lookups here.

File: layouts/shortcodes/release-data.html
Comment (sftim) [inline layouts/shortcodes/release-data.html:22]: We might want to the `, ` separator be a localized string lookup.

**Review by sftim — COMMENTED** [review]
@jimangel this looks really *really* promising. I also have a bit more feedback.

File: content/en/releases/patch-releases.md
Comment (kbhawkey) [inline content/en/releases/patch-releases.md:15]: Is this file auto-generated?

**Comment by kbhawkey** [comment]
Hi @jimangel . Some initial feedback:
The copied release pages seem okay. I guess the site offers a place to publish the release content for the release team.
How/when is the script run (release time)?
Would you consider including the release information in the docs content tree?
Why should the "Release Information" add to the `Version` drop down list? It does not really fit into this list.
What are the alternatives to providing ownership/approval to parts of this repository (another way of retrieving the release content and creating HTML content)?

File: static/_redirects
Comment (jimangel) [reply static/_redirects]: Updated, but please confirm we're talking about the same URL 😄

File: content/en/releases/patch-releases.md
Comment (jimangel) [reply content/en/releases/patch-releases.md:193]: Updated with generated warning at top of page for now. I would also be willing to remove the pages and starting with less content. Thanks for the heads up!

File: content/en/releases/release.md
Comment (jimangel) [reply content/en/releases/release.md:357]: Same resolution as above. Thanks!

File: content/en/releases/release.md
Comment (jimangel) [reply content/en/releases/release.md:364]: Same resolution as above. Thanks!

File: content/en/releases/patch-releases.md
Comment (jimangel) [reply content/en/releases/patch-releases.md:3]: This is somewhat of a "hack" to have Hugo look at /layouts/docs/ for it's templates. It allows me to use the same "Docs" 3-bar theme for a handful of release docs. It also reduces the need for me to create custom CSS.

**Comment by jimangel** [comment]
Thanks for the review @kbhawkey. I updated some of your comments inline and will answer the rest here.

> The copied release pages seem okay. I guess the site offers a place to publish the release content for the release team.

Exactly. This PR adds a location for release content and grants SIG Release co-ownership.

> How/when is the script run (release time)?

The script could be ran as needed at first (by release team or docs team). The ultimate goal would be to allow SIG Releases tooling to open the PR to update the data file. While the data is important, I don't believe it needs to be 100% automated to get started -- yet.

> Would you consider including the release information in the docs content tree?

I'm not sure what you're referencing here. Would you mind clarifying? As far as the current content, I would be ok with removing a lot of it if it improves the end user experience.

> Why should the "Release Information" add to the `Version` drop down list? It does not really fit into this list.

Totally agree! I REALLY wanted it to be a big header that said, "RELEASES" but when I set it up, it looked very strange next to "VERSIONS." I would imagine that would be hard to understand from a new user too. I don't love it where it is now (under the Versions menu) but it seems to make the most logical sense. I'm up for any additional placement / title! Any suggestions?

> What are the alternatives to providing ownership/approval to parts of this repository (another way of retrieving the release content and creating HTML content)?

There's alternative ways to run scripts (via Hugo, Netlify, Makefiles / shell) to capture content from URL endpoints. The concern with that is the reliance of data connectivity and increasing the complexity of our build dependencies. I would be up for alternative suggestions but I do feel like there is a very limited scope of access being opened up to SIG Release.

As always, thanks again for your review / attention to detail!

File: layouts/partials/navbar-version-selector.html
Comment (saschagrunert) [inline layouts/partials/navbar-version-selector.html:9]: Would it make sense to put the link on top of the list?

**Review by saschagrunert — COMMENTED** [review]
Generally I'm very supportive about this, glad to see it landing :heart: 

Thanks @jimangel! :pray:

**Review by justaugustus — APPROVED** [review]
From SIG Release, I'm +1 to merging and iterating on script cleanup, canonical doc locations, and layout, as-needed.

Great work, @jimangel (and @sftim, for your initial PR)!
/lgtm
/approve

cc: @kubernetes/sig-release-leads

**Comment by puerco** [comment]
This is awesome, great job Jim!!
/lgtm

**Comment by pmmalinov01** [comment]
This is amazing work. Thank you for it +1
/lgtm

File: content/en/releases/_index.md
Comment (celestehorgan) [reply content/en/releases/_index.md:10]: At this point I think we can drop the bit about k8s 1.18 and older:
```suggestion
The Kubernetes project maintains release branches for the most recent three minor releases ({{< skew latestVersion >}}, {{< skew prevMinorVersion >}}, {{< skew oldestMinorVersion >}}).  Kubernetes 1.19 and newer receive approximately 1 year of patch support.
```

File: data/releases/schedule.yaml
Comment (celestehorgan) [inline data/releases/schedule.yaml:1]: How does this  file get generated/added/updated? It might be helpful to add a comment indicating that (lest people add entries to an autogenerated file)

File: scripts/update-release-info.sh
Comment (celestehorgan) [inline scripts/update-release-info.sh:19]: Worth noting that when branch names flip over to `main`, this script will break

Comment (justaugustus) [reply scripts/update-release-info.sh:19]: @celestehorgan -- We'll obviate the need for this ahead of any branch renames.

File: data/releases/schedule.yaml
Comment (justaugustus) [reply data/releases/schedule.yaml:1]: @celestehorgan -- I can fast-follow with details on this once the PR merges.

Comment (jimangel) [reply data/releases/schedule.yaml:1]: I stole the file from [schedule-builder's readme](https://github.com/kubernetes/release/tree/master/cmd/schedule-builder). It looks like the file might be [manually](https://github.com/kubernetes/sig-release/commits/master/releases/schedule.yaml) updated (and same for patch-releases.md?) /cc @cpanato @xmudrii

File: content/en/releases/_index.md
Comment (jimangel) [reply content/en/releases/_index.md:10]: I don't think it hurts to leave it in for a little longer. I could be sold either way. 🤷

**Comment by jimangel** [comment]
/unhold

I just updated with everyone's actionable feedback. I left the open-ended questions posted for reference later. This is ready to `lgtm / merge` in my opinion.

**I would like +1/2 from a few more @kubernetes/sig-docs-leads** 🙏 ❤️

**Comment by justaugustus** [comment]
re-lgtm on behalf of SIG Release
/lgtm

**Review by irvifa — COMMENTED** [review]
/lgtm

File: content/en/releases/_index.md
Comment (bradtopol) [reply content/en/releases/_index.md:10]: I agree. I would leave it in a little longer.

**Comment by bradtopol** [comment]
I am very supportive of this.  This is outstanding work! Great job @jimangel!!!
/lgtm

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by sftim** [comment]
Given https://github.com/kubernetes/website/pull/27929#issuecomment-839237113 and other LGTMs
/approve

**Comment by sftim** [comment]
Issue #29711 requests enhancing this with release date information

**Pull request merged**
